cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -654,14 +657,13 @@ ReconciliationState createReconciliationState(Reconciliation reconciliation, Kaf
                                 ModelUtils.getCertificateValidity(clusterCaConfig),
                                 ModelUtils.getRenewalDays(clusterCaConfig),
                                 clusterCaConfig == null || clusterCaConfig.isGenerateCertificateAuthority(), clusterCaConfig != null ? clusterCaConfig.getCertificateExpirationPolicy() : null);
+                        this.clusterCa.initCaSecrets(clusterSecrets);
                         clusterCa.createRenewOrReplace(
                                 reconciliation.namespace(), reconciliation.name(), caLabels.toMap(),
                                 clusterCaCertLabels, clusterCaCertAnnotations,
                                 clusterCaConfig != null && !clusterCaConfig.isGenerateSecretOwnerReference() ? null : ownerRef,
                                 isMaintenanceTimeWindowsSatisfied(dateSupplier));
 
-                        this.clusterCa.initCaSecrets(clusterSecrets);
```

Comment:
because the `createRenewOrReplace` is going to call `hasCaCertGenerationChanged` which relies on Secret(s) storing the CA certificate generation to check if it's changed and needs to start the renewal procedure.